### PR TITLE
Add dependabot code ql container scans

### PR DESCRIPTION
**Pull request by GetTuh**
PR adding Dependabot and CodeQL scans before we pull in fork parents' changes. EISSWSCN-20070
After scanning it sets update-fork or needs-manual-check label. Setting update-fork label will update the fork repo automatically.

### CODE REVIEW

File: .github/workflows/check-workflow.yml [hunk 1]
```diff
@@ -14,38 +14,51 @@ jobs:
...
-    - uses: peter-murray/workflow-application-token-action@6a9741bf210c9a93370560585c908161170ecada # v2.1.0
-      name: Get token from GitHub App
-      if: ${{ vars.USE_GITHUB_APP == 'true' }}
-      id: get-workflow-token
-      with:
-        application_id: ${{ secrets.GH_AUTOMATION_ID }}
-        application_private_key: ${{ secrets.GH_AUTOMATION_PRIVATE_KEY }}
+      - uses: peter-murray/workflow-application-token-action@6a9741bf210c9a93370560585c908161170ecada # v2.1.0
+        name: Get token from GitHub App
+        if: ${{ vars.USE_GITHUB_APP == 'true' }}
```
Comment (r-derasari-asml) [inline .github/workflows/check-workflow.yml:19]: Do you need this var and if: condition? why not do this always?

File: .github/workflows/check-workflow.yml [hunk 2]
```diff
@@ -14,38 +14,51 @@ jobs:
...
-          $reposWithUpdates = $updateResult[-1]
-          Write-Host "Are there any updates available? [$($reposWithUpdates.Count -gt 0)]"
-          echo "::set-output name=update::$($reposWithUpdates.Count -gt 0)"
-        }
+      - id: update_step
+        shell: pwsh
+        name: Check for updates
+        run: |
+          $updateResult = (.\updater.ps1 -orgName ${{ github.repository_owner }} -userName "xxx" -PAT ${{ steps.get-token.outputs.token }} -issuesRepository ${{ github.repository }})
+
```
Comment (r-derasari-asml) [inline .github/workflows/check-workflow.yml:46]: 1. Don't use set-output
2. Let's try to use bash here? or python? for longetivity and maintenance of the code

File: .github/workflows/new-issue-notification.yml [hunk 3]
```diff
@@ -27,13 +27,13 @@ jobs:
...
-      name: Tag a user or team in an issue comment
-      with: 
-        issue: ${{ github.event.issue.number }}
-        team: ${{ vars.NOTIFICATION_TEAM }}
-        owner: ${{ github.repository_owner }}
-        repo: ${{ github.repository }}
-        GITHUB_TOKEN: ${{ steps.get-token.outputs.token }}
+        fi  
+        
+    # - uses: asml-actions/issue-comment-tag@v0.1.0
```
Comment (r-derasari-asml) [inline .github/workflows/new-issue-notification.yml:32]: Delete if not using, don't need to leave it commented :D

File: .github/workflows/validate-parent.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+name: Validate parent of fork
+
+on:
+  issues:
+    types: [opened]
```
Comment (r-derasari-asml) [inline .github/workflows/validate-parent.yml:5]: We have a lot of workflows running on "issue: opened", can we make a signal chain or something?

File: code-scan.js [hunk 5]
```diff
@@ -0,0 +1,203 @@
+const { Octokit } = require("@octokit/rest");
+const core = require("@actions/core");
+const { countReset } = require("console");
+const fs = require("fs");
+const token = process.argv[2];
+const repo = process.argv[3];
+const originalOwner = process.argv[4];
+const owner = "asml-actions-validation";
+console.log(`repo name is ${repo}, original owner ${originalOwner}`);
```
Comment (r-derasari-asml) [inline code-scan.js:9]: hoping that you have only kept the necessary log statements and nothing extra :D

File: code-scan.js [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    }
+    const response = await octokitFunctions[request](requestProperties);
+    console.log(`Function ${request} finished succesfully`);
+    return response.data;
+  } catch (error) {
+    console.log(`Failed to run ${request}: ${error.message}`);
+  }
+}
+
+async function putRequest(request) {
```
Comment (r-derasari-asml) [inline code-scan.js:51]: Do you _reallllly_ need a function here? probably more future proof but eh

File: codeql-analysis-check.yml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
```
Comment (r-derasari-asml) [inline codeql-analysis-check.yml:12]: Just reminding that this isn't in the workflows directory, i guess you probably know though/

File: .github/workflows/check-workflow.yml (hunk 1 above)
Comment (Rob-Bos) [reply .github/workflows/check-workflow.yml:19]: It's config setup from upstream. This change might warrant breaking the setup from upstream, but I'll have to do a full review first.

File: .github/workflows/check-workflow.yml (hunk 2 above)
Comment (Rob-Bos) [reply .github/workflows/check-workflow.yml:46]: again, coming from upstream, where it was chosen for longevity 😉

File: .github/workflows/check-workflow.yml [hunk 8]
```diff
@@ -14,38 +14,51 @@ jobs:
...
+          if ($null -ne $updateResult) {
+            $reposWithUpdates = $updateResult[-1]
+            Write-Host "Are there any updates available? [$($reposWithUpdates.Count -gt 0)]"
+            echo "::set-output name=update::$($reposWithUpdates.Count -gt 0)"
+            $updateResult | Set-Content -Path "updateResult.txt"
+          }
+
+      - uses: peter-murray/workflow-application-token-action@6a9741bf210c9a93370560585c908161170ecada # v2.1.0
```
Comment (Rob-Bos) [inline .github/workflows/check-workflow.yml:54]: Why get a new token? We already have one for this org on line 17

File: .github/workflows/validate-parent.yml (hunk 4 above)
Comment (Rob-Bos) [reply .github/workflows/validate-parent.yml:5]: Why? You run each workflow in isolation for a reason, so they need to trigger for a reason as well.

File: code-scan.js [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+
+async function wait(milliseconds) {
+  return new Promise((_resolve) => {
+    if (typeof milliseconds !== "number") {
+      throw new Error("milliseconds not a number");
+    }
+    setTimeout(() => _resolve("done!"), milliseconds);
+  });
+}
+
```
Comment (Rob-Bos) [inline code-scan.js:31]: Please cleanup :-D

File: code-scan.js [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+    console.log("Workflow file created successfully");
+  } catch (error) {
+    console.error("Error creating workflow file:", error);
+  }
+}
+
+async function triggerCodeqlScan(workflow_id, ref) {
+  console.log(`Trigger codeql scan`);
+  await octokit.rest.actions.createWorkflowDispatch({
```
Comment (Rob-Bos) [inline code-scan.js:115]: check the result of the dispatch for success should be 204, if not, bail

File: code-scan.js [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+    repo,
+    event: "workflow_dispatch",
+  });
+
+  const run_id = response.data.workflow_runs[0].id;
+
+  let status = "queued";
+  while (status != "completed") {
+    console.log(`Wait for scan to complete - Run id : ${run_id}`);
+    await wait(15000);
```
Comment (Rob-Bos) [inline code-scan.js:136]: This scan can potentially take a minute or more. Checking should be done in a loop until successful.

File: .github/workflows/validate-parent.yml [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Checkout the repo
+        uses: actions/checkout@v3
+
+      - uses: peter-murray/workflow-application-token-action@6a9741bf210c9a93370560585c908161170ecada # v2.1.0
+        name: Get token from GitHub App for validation
+        id: get-validation-token
+        with:
+          application_id: ${{ secrets.VALIDATION_APP_ID }}
+          application_private_key: ${{ secrets.VALIDATION_APP_TOKEN }}
+          organization: asml-actions-validation
```
Comment (Rob-Bos) [inline .github/workflows/validate-parent.yml:24]: Make this a `var` so we can merge this change upstream

File: .github/workflows/check-workflow.yml (hunk 8 above)
Comment (GetTuh) [reply .github/workflows/check-workflow.yml:54]: Leftovers from other implementation. I've deleted it

File: code-scan.js (hunk 11 above)
Comment (Rob-Bos) [reply code-scan.js:137]: nvm, it's already a look 🤦

File: .github/workflows/validate-parent.yml [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+on:
+  issues:
+    types: [opened]
+
+permissions:
+  issues: write
+  contents: read
+
+jobs:
+  validate-parent:
```
[the remark this answers is not in the document]
Comment (GetTuh) [reply .github/workflows/validate-parent.yml:13]: Done

File: code-scan.js (hunk 5 above)
Comment (GetTuh) [reply code-scan.js]: removed this log

File: code-scan.js (hunk 6 above)
Comment (GetTuh) [reply code-scan.js]: Adopted it so its also used by other function

**Pull request merged**